src/frequenz/client/common/pagination/pagination.py

```diff
@@ -0,0 +1,100 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Module to define the pagination used with the common client."""
+
+from __future__ import annotations  # required for constructor type hinting
+
+import logging
+from dataclasses import dataclass
+
+# pylint: disable=import-error, no-name-in-module
+from frequenz.api.common.v1.pagination_info_pb2 import (
+    PaginationInfo as PBPaginationInfo,
+)
+
+# pylint: disable=import-error, no-name-in-module
+from frequenz.api.common.v1.pagination_params_pb2 import (
+    PaginationParams as PBPaginationParams,
+)
+
+# Set up logging
+_logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class PaginationParams:
+    """
+    Parameters for paginating list requests.
+
+    Args:
+        page_size: The maximum number of results to be returned per request.
+        page_token: The token identifying a specific page of the list results.
+    """
+
+    page_size: int | None = None
+    page_token: str | None = None
+
+    @classmethod
+    def from_pb(cls, pagination_params: PBPaginationParams) -> PaginationParams:
```

Comment:
I would use `from_proto()` which is what we use in the SDK, I don't see a clear advantage of using `from_pb` and if we change it it will be a lot more work to update the SDK. Also please use `Self` (`from typing import Self`) as the return type, otherwise this won't work correctly if someone wants to subclass this for some reason.

```suggestion
    def from_proto(cls, pagination_params: PBPaginationParams) -> Self:
```